docs/examples/viz_tesseract.py

```diff
@@ -0,0 +1,186 @@
+A tesseract is a four-dimensional cube. A tesseract can
+be unfolded into eight cubes, Just as a cube can be unfolded into eight
+squares.
+"""
+
+###############################################################################
+# First, import some useful functions
+
+import numpy as np
+from fury import utils, actor, window
+from fury.ui import TextBlock2D
+
+###############################################################################
+# Let's define some variables and their descriptions:
+#
+# Use `wireframe = True` to show wireframe like representation of the tesseract
+# `wireframe = False` will render it with point actor on each vertex.
+
+wireframe = False
+
+# p_color: color of the point actor (default: (0, 0.5, 1, 1))
+# e_color: color of the line actor (default: (1, 1, 1, 1))
+# dtheta: change in `angle` on each iteration. It determines the "speed" of the
+#        animation. Increase dtheta to increase speed of rotation, It may
+#        result in less smoother rotation (default: 0.02)
+# angle: defines the angle to be rotated to perform the animation, It changes
+#        as we run the `callback` method later on. (initial value: 0)
+
+p_color = np.array([0, 0.5, 1, 1])
+e_color = np.array([1, 1, 1, 1])
+dtheta = 0.02
+angle = 0
+
+###############################################################################
+# Let's define vertices for our 4D cube, `verts4D` contains the coordinates of
+# our 4D tesseract.
+
+verts3D = np.array(
+    [[1, 1, 1],
+     [1, -1, 1],
+     [-1, -1, 1],
+     [-1, 1, 1],
+     [-1, 1, -1],
+     [1, 1, -1],
+     [1, -1, -1],
+     [-1, -1, -1]]
+)
+
+# We can use primitive.box alternatively to get the cube's 3-D vertices.
+
+u = np.insert(verts3D, 3, 1, axis=1)
+v = np.insert(verts3D, 3, -1, axis=1)
+verts4D = np.append(u, v, axis=0)
+
+###############################################################################
+# We define a `rotate4D` function that takes 4D matrix as parameter and rotates
+# it in XY plane (Z axis) and ZW plane (an imaginary axis), projects it to the
+# 3D plane so that we can render it in a scene.
+
+
+def rotate4D(verts4D):
+    cos = np.cos(angle)
+    sin = np.sin(angle)
+    rotation4d_xy = [[cos, -sin, 0, 0],
+                     [sin, cos, 0, 0],
+                     [0, 0, 1, 0],
+                     [0, 0, 0, 1]]
+    rotation4d_zw = [[1, 0, 0, 0],
+                     [0, 1, 0, 0],
+                     [0, 0, cos, -sin],
+                     [0, 0, sin, cos]]
+    distance = 2
+    projected_marix = []
+    for vert in verts4D:
+        rotated_3D = np.dot(rotation4d_xy, vert)
+        rotated_3D = np.dot(rotation4d_zw, rotated_3D)
+        w = 1/(distance - rotated_3D[3])
+        proj_mat4D = np.array(
+            [[w, 0, 0, 0],
+             [0, w, 0, 0],
+             [0, 0, w, 0]]
+        )
+
+        projeced_mat3D = np.dot(proj_mat4D, rotated_3D)
+        projected_marix.append(projeced_mat3D)  # vertices to be proj (3D)
```

Comment:
We recommend avoiding using a list of `np.array` because we observe memory issues.

Can you update your code? You can create another `np.array` with an additional dimension.

Reply:
Thanks for the review @skoudoro, Can I use np.zeros() and then change the elements or use np.array() then append the projected_mat3D ?

Reply:
both are ok